hazelcast/src/main/java/com/hazelcast/nio/serialization/FieldType.java

```diff
@@ -55,4 +71,32 @@ public static FieldType get(byte type) {
         return ALL[type];
     }
 
+    public boolean isArrayType() {
+        return type >= PORTABLE_ARRAY.type;
+    }
+
+    public FieldType getSingleType() {
+        if (isArrayType()) {
+            // GOTCHA: Wont' work if you add more types!!!
+            return get((byte) (getId() % TYPES_COUNT));
```

Comment:
Also easy to break, as far as I understand `FieldType` boils down to `(typeId, elementSize, multiplicity)`, so I would favor explicitly separating the `fieldTypeId` (e.g. 10 for `PORTABLE_ARRAY`) from the element `typeId` (which would be 0 for both `PORTABLE` and `PORTABLE_ARRAY`). Wdyt?